pkg/discovery/helper_test.go

```diff
@@ -84,3 +84,44 @@ func TestSortResources(t *testing.T) {
 		})
 	}
 }
+
+func TestFilteringByVerbs(t *testing.T) {
+	tests := []struct {
+		name         string
+		groupVersion string
+		res          *metav1.APIResource
+		expected     bool
+	}{
+		{
+			name:         "resource that supports list, create, get, delete",
```

Comment:
Can you do 1 more that has this same set of 4 but in a different order? Also please have at least 1 test where there are additional Verbs such as update, patch, and deletecollection.